internal/core/scale.go

```diff
@@ -26,7 +43,7 @@ func encodeProof(dnmp eth.DoubleNodeWithMerkleProof) string {
 	}
 
 	// pad the length
-	hex += "64"
+	hex += lenToHex(len(dnmp.Proof))
```

Comment:
Dynamic proof length

Reply:
This file name looks quite general, but the implementation looks quite domain specific. Are there scale codec go libraries for use, then we may do not need to try to implement here.

Reply:
[gossamer][0] supports `scale-codec`, but hard to use, Subscan also supports `scale-codec`, but only for decoding.

[0]: https://github.com/ChainSafe/gossamer